src/main/java/de/siphalor/amecs/VersionedLogicMethodHelper.java

```diff
@@ -0,0 +1,132 @@
+package de.siphalor.amecs;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+import org.apache.logging.log4j.Level;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.loader.api.SemanticVersion;
+import net.fabricmc.loader.api.VersionParsingException;
+
+@Environment(EnvType.CLIENT)
+public class VersionedLogicMethodHelper {
```

Comment:
Is this branch able to run on all the supported Minecraft versions (1.14-1.17)?

If not then this class should come in a separate pull request.

Reply:
I did not test it with versions other than 1.17.1. But I do not expect the logic for item dropping and mouse scrolling to change in versions down to 1.8. But that should be checked to be sure

Reply:
I just discovered that drop logic changed a little bit from mc 1.14 to 1.15 and is now bytecode incompatible with the newer versions. But I will fix it. It make 1.17 the base/compile version and call the 1.14 version's variant via reflection

Reply:
It actually is bytecode compatible so it is a very simple fix

Reply:
I added the 1.14 logic and call the method via reflection. But I did not test it with a mc 1.14 version yet. You could test it when backporting to 1.14